### PR TITLE
11. GET /api/reviews (queries)

### PR DESCRIPTION
**Pull request by dafyddhenke**
Card id: https://trello.com/c/P5SJBrTl

- can filter by category (default none)
- can order by asc and desc (default desc)
- can sort any column (default date)
- can use these all together, as well.
- included tests for valid and invalid attempts to use it

I struggled with the tests for this one. For example, should I test that I can filter *every* category? Ultimately, they are all just different strings. 

As well, should I try every combination of everything? I can see the value in it, but it would also take me some time. I've tried to strike some balance.

### CODE REVIEW

File: __tests__/reviews.controller.test.js [hunk 1]
```diff
@@ -57,15 +71,211 @@ describe("GET /api/reviews", () => {
...
+      .then((response) => {
+        const categoryArray = response.body;
+        const correctValues = categoryArray.every((game) => {
+          return game.category === "social deduction";
+        });
+        expect(categoryArray.length === 11).toBe(true);
+        expect(correctValues).toBe(true);
+      });
+  });
+  test("GET /api/reviews should give an appropriate error message when attempting to filter by category that does not exist", () => {
```
Comment (HannahBurks) [inline __tests__/reviews.controller.test.js:271]: Excellent testing Dafydd, however  I would like to see one more test for Status 200, valid category query, but has no reviews responds with an empty array of reviews, e.g. ?category=children's games

**Review by HannahBurks — CHANGES_REQUESTED** [review]
Just one extra test to add - once done please send another PR :)

File: __tests__/reviews.controller.test.js (hunk 1 above)
Comment (dafyddhenke) [reply __tests__/reviews.controller.test.js:280]: Done.
